chef/cookbooks/crowbar/templates/default/crowbar-router-migration.sh.erb

```diff
@@ -0,0 +1,29 @@
+#!/bin/bash
+#
+# This script starts the upgrade of the node server to the latest version of
+# system and Cloud product.
+
+LOGFILE=/var/log/crowbar/node-upgrade.log
+mkdir -p "`dirname "$LOGFILE"`"
+exec >>"$LOGFILE" 2>&1
+
+echo "Executing $BASH_SOURCE"
+
+set -x
+
+source /root/.openrc
+routers=$(neutron router-list -f value -c id)
+for router in $routers; do
+    # delete old router assignment to agent
+    agent=$(neutron l3-agent-list-hosting-router $router -f value -c id)
+    neutron l3-agent-router-remove $agent $router
+
+    # assign router to agent in upgraded node
+    upgraded_agent=$(neutron agent-list --binary neutron-l3-agent --host $(hostname) -f value -c id)
+    neutron l3-agent-router-add $upgraded_agent $router
+
+    # delete old routers namespaces. Unassignment didn't delete the namespaces
+    old_host=$(neutron agent-show $agent -f value -c host)   # check if we can use '$' inside '$()'
```

Comment:
I don't understand. Why can't you do:
```
agent=$(neutron l3-agent-list-hosting-router $router -f value -c id)
old_host=$(neutron agent-show $agent -f value -c host)
```
at the beginning?

The reason I'm suggesting this is that if we can move all the neutron calls (that are taking some time) early on, then the migration will be faster.